### ISSUE TITLE
Unable to run a map reduce job in a cluster different from where Oozie is running.

### ISSUE THREAD

**Issue by deepikakhera** [YahooArchive/oozie#616]
If a map reduce job is configured to run on a cluster different from the cluster on which Oozie is running, it fails with error.

**Comment by deepikakhera** [comment]
Closing this issue as it is a duplicate of issue 527.

**Comment by mislam77-zz** [comment]
Hi,
Oozie is primarily tested for the same cluster. Therefore there could be some 
obvious bug. However, if you can give more details with LauncherMapper log along 
with oozie.log, it could be easy to troubleshoot.

Oozie team at yahoo is now working to certify multiple cluster mode where oozie 
should work in any combination like yours. We expected to include the feature in 
3.0.x patch release.

Regards,
Mohammad

---

From: deepikakhera 
reply@reply.github.com
To: mislam77@yahoo.com
Sent: Mon, March 21, 2011 9:30:07 PM
Subject: [GitHub] Unable to run a map reduce job in a cluster different from 
where Oozie is running. [yahoo/oozie GH-616]

If a map reduce job is configured to run on a cluster different from the cluster 
on which Oozie is running, it fails with error.

## 

Reply to this email directly or view it on GitHub:
https://github.com/yahoo/oozie/issues/616

**Comment by deepikakhera** [comment]
Hi Mohammad,

I needed the feature now, so I did a quick patch to get it working (did it for version 2.2 for oozie. Now ,  I plan to use it for 2.3). Basically the issue was that JavaActionExecutor.getActionFileSystem(Context context, Element actionXml), is not using the filesystem from actionXml (which has the configuration for the second job that is to be run in the other cluster).Instead it uses the filesystem of workflow.

The exception I got (when tested with version 2.2) is:

IllegalArgumentException: Wrong FS: hdfs://cluster1namenode:9000/user/Deepika/oozie-root/0000006-110201204059448-oozie-root-W/cluster1job1--map-reduce, expected: hdfs://cluster2namenode:9000.

The workflow directory here is cluster1. The exception is thrown when the workflow tries to run a map reduce job in cluster2.

In my patch, I modified the getActionFileSystem method to use the filesystem from the actionXml for "map reduce jobs", when that is present.

I can submit the patch, if you think it will be useful - until the permanent fix is out in version 3.0.

Thanks,
Deepika
